src/overlay_16_022014A0.c

```diff
@@ -0,0 +1,546 @@
+    bl ov16_02201528
+    strh r0, [r4, #2]
+    ldrb r0, [r4, #1]
+    cmp r0, #5
+    bne _022017E4
+    add r0, r4, #0
+    bl ov16_02201578
+    ldrh r1, [r4, #2]
+    mul r0, r1
+    strh r0, [r4, #2]
+_022017E4:
+    ldrb r0, [r4, #1]
+    cmp r0, #0
+    beq _022017EE
+    cmp r5, #0
+    bne _0220179A
+_022017EE:
+    ldr r0, [sp]
+    add r4, #0xc
+    add r0, r0, #1
+    str r0, [sp]
+    cmp r0, #4
+    blt _0220176C
+    pop {r3, r4, r5, r6, r7, pc}
+}
+
+void ov16_022017FC(struct GF_RTC_DateTime *dest, struct GF_RTC_DateTime src);
+void ov16_022017FC(struct GF_RTC_DateTime *dest, struct GF_RTC_DateTime src) {
+    dest->date = src.date;
+    dest->time = src.time;
+}
+
+void ov16_02201820(struct GF_RTC_DateTime *datetime, RTCDate *date, RTCTime *time);
+void ov16_02201820(struct GF_RTC_DateTime *datetime, RTCDate *date, RTCTime *time) {
+    *date = datetime->date;
+    *time = datetime->time;
+}
+
+NARC *ov16_02201840(HeapID heapId) {
+    return NARC_New(NARC_a_0_6_6, heapId);
+}
+
+UnkStruct_a066 *ov16_0220184C(NARC *narc, s32 fileId, HeapID heapId) {
+    return NARC_AllocAndReadWholeMember(narc, fileId, heapId);
+}
+
+void ov16_02201854(NARC *narc) {
+    NARC_Delete(narc);
+}
+
+u16 ov16_0220185C(UnkStruct_a066 *unk, u32 attr) {
+    switch (attr) {
+        case 0:
+            return unk->unk0;
+        case 1:
+            return unk->unk2;
+        case 2:
+            return unk->unk3;
+        case 3:
+            return unk->unk4;
+        case 4:
+            return unk->unk5;
+        case 5:
+            return unk->unk6;
+        case 6:
+            return unk->unk7;
+        case 7:
+            return unk->unk8;
+        case 8:
+            return unk->unk9;
+        case 9:
+            return unk->unkA;
+        case 10:
+            return unk->unkB;
+        default:
+            return 0;
+    }
+}
+
+u16 ov16_022018B4(u16 berryId);
+u16 ov16_022018B4(u16 berryId) {
+    if (berryId == 0) {
+        return ITEM_NONE;
+    }
+
+    return berryId + FIRST_BERRY_IDX - 1;
+}
+
+u16 ov16_022018C4(u16 itemId);
+u16 ov16_022018C4(u16 itemId) {
+    if (itemId == ITEM_NONE) {
+        return 0;
+    }
+
+    return itemId - FIRST_BERRY_IDX + 1;
+}
+
+u16 ov16_022018D4(u16 mulchId);
+u16 ov16_022018D4(u16 mulchId) {
+    if (mulchId == 0) {
+        return ITEM_NONE;
+    }
+
+    return mulchId + ITEM_GROWTH_MULCH - 1;
+}
+
+u32 ov16_022018E4(u16 itemId);
+u32 ov16_022018E4(u16 itemId) {
+    if (itemId == ITEM_NONE) {
+        return 0;
+    }
+
+    return itemId - ITEM_GROWTH_MULCH + 1;
+}
+
+// total berry quantity
+u16 ov16_022018F0(Bag *bag, HeapID heapId);
+u16 ov16_022018F0(Bag *bag, HeapID heapId) {
+    s32 i;
+    u16 total = 0;
+    for (total = 0, i = 0; i < NUM_BAG_BERRIES; i++) {
+        total += Bag_GetQuantity(bag, FIRST_BERRY_IDX + i, heapId);
+    }
+    return total;
+}
+
+// total mulch quantity
+u16 ov16_0220191C(Bag *bag, HeapID heapId);
+u16 ov16_0220191C(Bag *bag, HeapID heapId) {
```

Comment:
Rename this + other functions you've left comments on so we can have them documented?